### PR TITLE
GPL-659 & GPL-nnn filter high ct out of reports and Lighthouse Sentinel

### PR DESCRIPTION
**Pull request by KatyTaylor**
I had to change a bunch of unrelated tests because I added to the 'samples' fixture data.

Most important changes for this story are in:
- helpers/reports.py (GPL-659)
- tests/helpers/test_reports_helpers.py (GPL-659)
- helpers/plates.py (GPL-nnn)
- tests/helpers/test_plates_helpers.py (GPL-nnn)
- constants (both)
- fixture_data.py (both)

### CODE REVIEW

File: lighthouse/constants.py [hunk 1]
```diff
@@ -10,9 +10,14 @@
...
 
 # MLWH lighthouse samples table field names
 MLWH_LH_SAMPLE_ROOT_SAMPLE_ID ="root_sample_id"
 MLWH_LH_SAMPLE_COG_UK_ID = "cog_uk_id"
 MLWH_LH_SAMPLE_RNA_ID = "rna_id"
-MLWH_LH_SAMPLE_RESULT = "result"
\ No newline at end of file
+MLWH_LH_SAMPLE_RESULT = "result"
+
+CT_VALUE_LIMIT = 30
```
Comment (andrewsparkes) [inline lighthouse/constants.py]: Unsure how datatypes work in python, we want a float as the values are e.g. 27.2631537

File: tests/data/fixture_data.py [hunk 2]
```diff
@@ -117,6 +120,93 @@
...
+    },
+    { # positive, with low Ct values
+        FIELD_COORDINATE: "E01",
+        FIELD_SOURCE: "test1",
+        FIELD_RESULT: "Positive",
+        FIELD_PLATE_BARCODE: "123",
+        FIELD_COG_BARCODE: "nop",
+        FIELD_ROOT_SAMPLE_ID: "MCM005",
+        FIELD_RNA_ID: "rna_1",
+        FIELD_CQ_1: 5,
```
Comment (andrewsparkes) [inline tests/data/fixture_data.py:140]: Cq values are floats with about 8dp from the examples, so maybe make the tests use more realistic float numbers e.g. 5.32563219 etc.

File: lighthouse/constants.py [hunk 3]
```diff
@@ -10,9 +10,14 @@
 FIELD_PLATE_BARCODE = "plate_barcode"
 FIELD_COG_BARCODE = "cog_barcode"
 FIELD_DATE_TESTED = "Date Tested"
+FIELD_CQ_1 = "ch1_cq"
```
Comment (andrewsparkes) [inline lighthouse/constants.py:13]: I named these after the fields in crawler, so FIELD_CH1_CQ, FIELD_CH1_RESULT, FIELD_CH1_TARGET etc., dunno if it's worth keeping things consistent?
Also there are 4 channels, assume 4 is not relevant here?

File: lighthouse/helpers/reports.py [hunk 4]
```diff
@@ -224,7 +228,31 @@ def get_all_positive_samples(samples):
     logger.debug("Getting all positive samples")
     # filtering using case insensitive regex to catch "Positive" and "positive"
     results = samples.find(
-        filter={FIELD_RESULT: {"$regex": "^positive", "$options": "i"}},
+        filter={
+            "$and": [
+                {
+                    FIELD_RESULT: { "$regex": "^positive", "$options": "i" }
```
Comment (andrewsparkes) [inline lighthouse/helpers/reports.py]: Possible we may have 'Detected' which = 'Positive' as well (sigh).

File: lighthouse/constants.py [hunk 5]
```diff
@@ -10,9 +10,36 @@
...
 
 # MLWH lighthouse samples table field names
 MLWH_LH_SAMPLE_ROOT_SAMPLE_ID ="root_sample_id"
 MLWH_LH_SAMPLE_COG_UK_ID = "cog_uk_id"
 MLWH_LH_SAMPLE_RNA_ID = "rna_id"
-MLWH_LH_SAMPLE_RESULT = "result"
\ No newline at end of file
+MLWH_LH_SAMPLE_RESULT = "result"
+
+CT_VALUE_LIMIT = 30
```
Comment (andrewsparkes) [inline lighthouse/constants.py:23]: float? examples have 8dp e.g. 30.0
how do datatypes work in python

File: lighthouse/constants.py [hunk 6]
```diff
@@ -10,9 +10,36 @@
...
+
+POSITIVE_SAMPLES_MONGODB_FILTER = {
+  FIELD_RESULT: {
+    "$regex": "^positive", "$options": "i"
```
Comment (andrewsparkes) [inline lighthouse/constants.py:27]: May also be 'Detected' (= Positive)

File: lighthouse/constants.py (hunk 3 above)
Comment (KatyTaylor) [reply lighthouse/constants.py:13]: Yep OK will change. Don't think we need 4 here, as not filtering on it

File: lighthouse/constants.py (hunk 5 above)
Comment (KatyTaylor) [reply lighthouse/constants.py:23]: > Python fully supports mixed arithmetic: when a binary arithmetic operator has operands of different numeric types, the operand with the “narrower” type is widened to that of the other, where integer is narrower than floating point, which is narrower than complex. A comparison between numbers of different types behaves as though the exact values of those numbers were being compared.

 (https://docs.python.org/3/library/stdtypes.html)

Seems OK :) I'll change the numbers in the tests to floats

File: lighthouse/constants.py (hunk 6 above)
Comment (KatyTaylor) [reply lighthouse/constants.py:27]: Are we waiting to hear back from Rich on that one?
If there have been 'Detected' ones coming through in the past, we won't have been cherrypicking them :(

**Review by andrewsparkes — DISMISSED** [review]

**Review by andrewsparkes — APPROVED** [review]

**Pull request merged**